lib/directory-tree.js

```diff
@@ -30,32 +30,47 @@ function directoryTree (path, options, onEachFile) {
 
 	// Skip if it matches the exclude regex
 	if (options && options.exclude && options.exclude.test(path))
-		return null;  
+		return null;
 
 	if (stats.isFile()) {
-		
+
 		const ext = PATH.extname(path).toLowerCase();
-		
+
 		// Skip if it does not match the extension regex
 		if (options && options.extensions && !options.extensions.test(ext))
 			return null;
 
 		item.size = stats.size;  // File size in bytes
 		item.extension = ext;
 		item.type = constants.FILE;
+
+		if (options && options.attributes) {
+			options.attributes.forEach((attribute) => {
+				item[attribute] = stats[attribute];
+			});
+		}
+
 		if (onEachFile) {
-			onEachFile(item, PATH);
+			onEachFile(item, PATH, stats);
 		}
 	}
 	else if (stats.isDirectory()) {
 		let dirData = safeReadDirSync(path);
 		if (dirData === null) return null;
-		
+
+		if (options && options.attributes) {
+			options.attributes.forEach((attribute) => {
+				item[attribute] = stats[attribute];
+			});
+		}
 		item.children = dirData
 			.map(child => directoryTree(PATH.join(path, child), options, onEachFile))
 			.filter(e => !!e);
 		item.size = item.children.reduce((prev, cur) => prev + cur.size, 0);
 		item.type = constants.DIRECTORY;
+		if (onEachDirectory) {
```

Comment:
What if we keep one single callback and the user has to check the item's type inside? Since you'll get the stats now.

Reply:
I like that even better. Will push up a commit with that soon